core/src/hlist.rs

```diff
@@ -1438,11 +1459,13 @@ where
 impl<H, Tail> Into<Vec<H>> for HCons<H, Tail>
 where
     Tail: Into<Vec<H>> + HList,
+    <Tail as HList>::Len: Add<UInt<UTerm, B1>>,
+    <<Tail as HList>::Len as Add<UInt<UTerm, B1>>>::Output: Unsigned,
 {
     fn into(self) -> Vec<H> {
         let h = self.head;
         let t = self.tail;
-        let mut v = Vec::with_capacity(<Self as HList>::LEN);
+        let mut v = Vec::with_capacity(<<Self as HList>::Len as Unsigned>::USIZE);
```

Comment:
One option is to return an array instead of a vector, because now the length is fundementally know at compile time. `GenericArray` could be used, making this a no_std compatable conversion